test/connect.js

```diff
@@ -417,4 +417,40 @@ QUnit.module('@connect with ObserveObject', () => {
 		assert.equal(getTextFromElement(parentDiv), 'abAcdBeCfghi', 'abAcdBeCfghi');
 
 	});
+
+	QUnit.asyncTest('should only use splice on arrays passed into children not objects', (assert) => {
+		class FooList extends ObserveArray {
+			constructor(initialValue, bar) {
+				super(initialValue);
+				this.bar = bar;
+			}
+		}
+		
+		function makeFooArray(bar) {
+			const fooArray = [ 1, 2, 3 ];
+			fooArray.bar = bar;
+			return fooArray;
+		}
+
+		class ParentStore extends ObserveObject {
+			constructor(props) {
+				super(props);
+				setTimeout((value) => {
+					this.fooObject = new FooList([{ foo0: 'foo0' }], 'bam');
+					this.fooArray = makeFooArray('bam');
+					assert.equal(getTextFromElement(parentDiv), 'bambar', 'bambar');
+					QUnit.start();
```

Comment:
We probably shouldn't use these old, no-longer-documented, Qunit APIs (`QUnit.asyncTest`, `QUnit.start()`, ...etc)  in our tests.
There are new and documented QUnit methods to do these kinds of async tests
https://api.qunitjs.com/assert/async